Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -136,13 +136,44 @@ static Gfx::StandardCursor cursor_css_to_gfx(Optional<CSS::Cursor> cursor)
     }
 }
 
+// https://drafts.csswg.org/cssom-view/#dom-mouseevent-offsetx
 static CSSPixelPoint compute_mouse_event_offset(CSSPixelPoint position, Layout::Node const& layout_node)
 {
-    auto top_left_of_layout_node = layout_node.first_paintable()->box_type_agnostic_position();
```

Comment:
box_type_agnostic_position() returns position without transformations being applied. what does this change improve?

Reply:
Given this page: (a 1px square div, scaled up to visually be a 100px square)
```html
<style>
  div {
    width: 1px;
    height: 1px;
    background-color: gray;
    position: absolute;
    top: 100px;
    left: 100px;
    transform: scale(100);
  }
</style>

<div onClick="console.log(event.offsetX, event.offsetY);"></div>
```

Without this change, mouse events on this div will have `offsetX` and `offsetY` values ranging from -50 to 50, cause they'd just be relative to the element's top left corner in 2D, ignoring its transforms.
With this change, the `offsetX` and `offsetY` values will range from 0 to 1.

When the spec says to ignore the element's transforms here, what it means is that you take the local XY coordinate that was visually clicked on the div from the div's perspective, and then you do not transform that back to be in screen space coordinates.

Personally, I think the spec's wording is kinda backwards and ambiguous here, but that's the only sensible way for offsetX and Y to work, and also what other browsers implement. (Namely, Chrome and Firefox, I do not have Safari to test)

Reply:
Got it, I confused that with CSS transform. Could you add a test for this improvement? We have internals.click() to emulate click event in tests.

Reply:
I have now added a test, and it also covers #2364 since it just tests all the mouse position values on a transformed div, with the page scrolled down a bit.

Reply:
The test has been updated to only cover the offsetX/Y-on-transformed-element behavior, and the rest of it has been moved to #2364